--- trunk/lib/device.c.old	2006-06-24 17:36:59.248122032 +0200
+++ trunk/lib/device.c	2006-06-24 17:38:52.999829144 +0200
@@ -1115,6 +1115,8 @@
   else
     {
       if ((strncmp (dev, "/dev/ataraid/", 13) == 0) ||
+         (strncmp (dev, "/dev/ida/", 9) == 0) ||
+         (strncmp (dev, "/dev/cciss/", 11) == 0) ||
          (strncmp (dev, "/dev/rd/", 8) == 0))
         strcpy (dev + strlen(dev), "p");
     }
